lcls-twincat-common-components/Library/POUs/FB_FDQ_FlowMeter.TcPOU

```diff
@@ -0,0 +1,22 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<TcPlcObject Version="1.1.0.1">
+  <POU Name="FB_FDQ_FlowMeter" Id="{b5b12a66-8d5f-455f-b8d1-7dfb923b9d24}" SpecialFunc="None">
+    <Declaration><![CDATA[FUNCTION_BLOCK FB_FDQ_FlowMeter EXTENDS FB_AnalogInput
+VAR_INPUT
+END_VAR
+VAR_OUTPUT
+    {attribute 'pytmc' := '
+        pv: FWM
+        field: EGU lpm
+    '}
+    fFlow : LREAL;
+END_VAR
+VAR
+END_VAR
+]]></Declaration>
+    <Implementation>
+      <ST><![CDATA[SUPER^(iTermBits:=15, fTermMax:=60, fTermMin:=0);
```

Comment:
Note that, implemented like this using `EXTENDS` and `SUPER`, you now have an FB that accepts all of the inputs from https://github.com/pcdshub/lcls-twincat-general/blob/master/LCLSGeneral/LCLSGeneral/POUs/Hardware/FB_AnalogInput.TcPOU and ignores them. This could be confusing to other engineers later, and makes it so all consumers of `FB_FDQ_FlowMeter` are forced into the same settings until we revisit this FB.

In general I don't think the inheritance here makes your implementation cleaner.

Reply:
I see your point here about not passing the inputs. We should include that. The pro to inheriting here is that if instead I only wrap the `FB_AnalogInput` I ended up having to write double level pragmas at the user level:
```
    .fbFlowMeter.iRaw := TIIB[EP3174-FWM-E2]^AI Standard Channel 1^Value
    '}
    ST4K4: FB_MotionPneumaticActuator;
```
ends up like:
```
    .fbFlowMeter.fbFlowMeter.iRaw := TIIB[EP3174-FWM-E2]^AI Standard Channel 1^Value
    '}
    ST4K4: FB_MotionPneumaticActuator;
```
Maybe this okay and worth avoiding the longterm headaches of inheritance stacks.

Open to other implementation ideas. The main goal here was only to have 1 spot to update values like pragma fields such as EGU and thresholds etc. Maybe the sensor itself actually needs a more resolute scope. Meaning, FDQ_KB_FlowMeter for example, where all these meters will likely all have the same EGU and Alarm Thresholds etc.

Reply:
Having shorter pragmas is a reasonable and good thing to try to do. I guess without inheritance if you want to achieve the same you'd need to add your own iRaw which seems wasteful, so maybe as-is is the best decision overall (it's hard to say).

None of these are blocking comments btw, just some food for thought. I'm curious what the Beckhoff-recommended way to handle inputs in the FB subclass is.